make test on M1 Mac

make test fnum=2 の前半の最後の方で，xargs が下記のようなエラーを出して失敗している
(M1 Mac, macOS 14.0, zsh)
```
（途中略）
hylagi ../examples/roof_bouncing.hydla finished
hylagi ../examples/navigation.hydla finished
hylagi ../examples/sawtooth_wave.hydla finished
hylagi ../examples/simple_curling.hydla finished
xargs: command line cannot be assembled, too long
hylagi ../examples/simple_water_tanks.hydla finished
make: *** [test] Error 1
```

make test で, xargsが下記のエラーを出して失敗する.  改善前よりも手前でエラーを出している. 
(M1 Mac, Intel Mac, macOS 15.0, zsh)
```
testing system_test in 1 parallel...
hylagi ../examples/2d_billiard.hydla finished
hylagi ../examples/3d_bouncing_ball.hydla finished
hylagi ../examples/DC-DC.hydla finished
hylagi ../examples/billiard.hydla finished
hylagi ../examples/billiard_quad.hydla finished
hylagi ../examples/bouncing_ball_box.hydla finished
hylagi ../examples/bouncing_particle.hydla finished
xargs: command line cannot be assembled, too long
make: *** [test] Error 1
```

--- a/system_test/system_test.sh
+++ b/system_test/system_test.sh
@@ -14,10 +14,18 @@ if [ -z "$fnum" ]; then
 fi
 echo " in $fnum parallel..."
 
-ls ../examples/*.hydla | xargs -P $fnum -I {} bash -c '../bin/hylagi {} &> /dev/null && printf "../bin/hylagi %s \033[32m%s\033[m\n" {} "finished" || (printf "../bin/hylagi %s \033[31m%s\033[m\n" {} "failed"; exit 255)'
+function test(){
+  ../bin/hylagi $1 &> /dev/null \
+    && printf "../bin/hylagi %s \033[32m%s\033[m\n" $1 "finished" \
+    || (printf "../bin/hylagi %s \033[31m%s\033[m\n" $1 "failed"; \
+        exit 255)
+}
+export -f test
+
+ls ../examples/*.hydla | xargs -P $fnum -I {} bash -c "test {}"
 printf "%s \033[32m%s\033[m\n" "exec examples" "succeeded"
 
-ls ../check_examples/*.hydla | xargs -P $fnum -I {} bash -c '../bin/hylagi {} &> /dev/null && printf "../bin/hylagi %s \033[32m%s\033[m\n" {} "finished" || (printf "../bin/hylagi %s \033[31m%s\033[m\n" {} "failed"; exit 255)'
+ls ../check_examples/*.hydla | xargs -P $fnum -I {} bash -c "test {}"
 printf "%s \033[32m%s\033[m\n" "exec check_examples" "succeeded"
 
 ls hydat/*.hydat | xargs -P $fnum -I {} python3 compare_hydat.py {} {}.master

